### PR TITLE
Assignment Schedule

### PR DESCRIPTION
**Pull request by chrisjpurdy**
This implements a new page to manage assignments based on their start and due dates. It's accessed via `/assignment_schedule`.

The idea is that this is released to live and anyone with the link can access it, and then we add it to the navbar later on. It is probably going to remain separate to the set assignments page since it represents a different part of the assignment workflow.

I have yet to add buttons to add gameboard from books and "boards for lessons", but this shouldn't hold back merging because the whole page is a hidden beta feature subject to change. There also isn't yet functionality to modify your current assignments - this can also be done later since it requires an API change (probably a new endpoint).

There are a couple of changes to gameboards and dates which are transient bug fixes - the commits containing these are separate from the new additions.

### CODE REVIEW

File: src/app/services/assignments.ts [hunk 1]
```diff
@@ -14,8 +14,8 @@ export const filterAssignmentsByStatus = (assignments: AssignmentDTO[] | undefin
     const fourWeeksAgo = new Date(now.valueOf() - (4 * 7 * 24 * 60 * 60 * 1000));
     // Midnight five days ago:
     const fiveDaysAgo = new Date(now);
-    fiveDaysAgo.setDate(now.getDate() - 5);
-    fiveDaysAgo.setHours(0, 0, 0, 0);
+    fiveDaysAgo.setUTCDate(now.getDate() - 5);
+    fiveDaysAgo.setUTCHours(0, 0, 0, 0);
```
Comment (Morpheu5) [inline src/app/services/assignments.ts:18]: There are other similar changes and the whole PR seems to adopt UTC for hours. It may be fine but I wonder if this is going to have weird effects with the fact that we schedule assignments at a certain time in the timezone of the browser instead of UTC.

File: src/app/components/handlers/ShowLoadingQuery.tsx [hunk 2]
```diff
@@ -1,32 +1,80 @@
...
+    return <Alert color={"warning"}>
+        {title ?? "Error fetching data from server"}: {errorDetails.message}
+        {errorDetails.status ? <><br/>Status code: {errorDetails.status}</> : ""}
+        <br/>
+        You may want to refresh the page, or <a href={`mailto:${WEBMASTER_EMAIL}`}>email</a> us if
+        this continues to happen.
+        Please include in your email the name and email associated with this
+        Isaac {SITE_SUBJECT_TITLE} account, alongside the details of the error given above.
+    </Alert>;
+};
```
Comment (Morpheu5) [inline src/app/components/handlers/ShowLoadingQuery.tsx:26]: I'm assuming this (and related changes) isn't strictly related to this PR. Any chance these changes can be separated out into another PR? OK if not, it just feels a bit unclean, but I think we can survive.

**Review by Morpheu5 — COMMENTED** [review]
Nice job overall, the UI looks very neat and seems to do what it's supposed to do. I'm just going to leave those two comments there.

File: src/app/services/assignments.ts (hunk 1 above)
Comment (chrisjpurdy) [reply src/app/services/assignments.ts]: Sorted this 👍

**Review by Morpheu5 — APPROVED** [review]

File: src/app/components/handlers/ShowLoadingQuery.tsx (hunk 2 above)
Comment (chrisjpurdy) [reply src/app/components/handlers/ShowLoadingQuery.tsx:26]: This is not _strictly_ related to this PR, but I had to build it in order to show errors and loading status for the assignment schedule in a clean way. I'd rather not remove it since I'd end up having to reimplement a version of it in `AssignmentSchedule` anyway
